chat/rich-links/src/RichLink/RichLink.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { forwardRef } from 'react';
+
+import Card from '@leafygreen-ui/card';
+import { cx } from '@leafygreen-ui/emotion';
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+import { HTMLElementProps } from '@leafygreen-ui/lib';
+import { Body } from '@leafygreen-ui/typography';
+
+import {
+  badgeAreaStyles,
+  baseStyles,
+  imageBackgroundStyles,
+  richLinkTextClassName,
+  themeStyles,
+} from './RichLink.styles';
+import {
+  RichLinkBadgeControlProps,
+  type RichLinkProps,
+} from './RichLink.types';
+import { RichLinkBadge } from './RichLinkBadge';
+import { richLinkVariants } from './RichLinkVariants';
+
+type DivProps = HTMLElementProps<'div', never>;
+
+export const RichLink = forwardRef<HTMLAnchorElement, RichLinkProps>(
+  ({ darkMode: darkModeProp, ...props }, ref) => {
+    const { darkMode, theme } = useDarkMode(darkModeProp);
+
+    if ('variant' in props) {
+      const { variant, ...variantProps } = props;
+      const RichLinkVariant = richLinkVariants[variant];
+      return <RichLinkVariant darkMode={darkMode} {...variantProps} />;
+    }
+
+    const badgeDefaults: Partial<RichLinkBadgeControlProps> = {
+      badgeGlyph: undefined,
+      badgeLabel: undefined,
+      badgeColor: undefined,
+    };
+
+    const {
+      children,
+      imageUrl,
+      badgeGlyph,
+      badgeLabel,
+      badgeColor,
+      href,
+      ...anchorProps
+    } = {
+      ...badgeDefaults,
+      ...props,
+    };
+
+    const showBadge = badgeLabel !== undefined;
+
+    const showImageBackground = (imageUrl?.length ?? -1) > 0;
+
+    return (
+      <Card
```

Comment:
I just noticed that there are no focus styles, and I think it's because the [`contentStyle`](https://github.com/mongodb/leafygreen-ui/blob/main/packages/card/README.md#properties) prop is missing. Can you add it?

Reply:
Ah that's my bad - in the refactor I forgot to pass `href` to the card so the focus selector wasn't appearing. Fixed the href and now focus states are working properly again.



https://github.com/mongodb/leafygreen-ui/assets/15657698/fbc95ee4-8528-4f6c-aa98-a272b8d88e82